guacamole/src/main/java/org/glyptodon/guacamole/net/basic/GuacamoleSession.java

```diff
@@ -76,80 +69,110 @@
     private long lastAccessedTime;
     
     /**
-     * Creates a new Guacamole session associated with the given user context.
+     * Creates a new Guacamole session associated with the given
+     * AuthenticatedUser and UserContexts.
      *
      * @param environment
      *     The environment of the Guacamole server associated with this new
      *     session.
      *
-     * @param credentials
-     *     The credentials provided by the user during login.
+     * @param authenticatedUser
+     *     The authenticated user to associate this session with.
      *
-     * @param userContext
-     *     The user context to associate this session with.
+     * @param userContexts
+     *     The List of UserContexts to associate with this session.
      *
      * @throws GuacamoleException
      *     If an error prevents the session from being created.
      */
-    public GuacamoleSession(Environment environment, Credentials credentials,
-            UserContext userContext) throws GuacamoleException {
+    public GuacamoleSession(Environment environment,
+            AuthenticatedUser authenticatedUser,
+            List<UserContext> userContexts)
+            throws GuacamoleException {
         this.lastAccessedTime = System.currentTimeMillis();
-        this.credentials = credentials;
-        this.userContext = userContext;
+        this.authenticatedUser = authenticatedUser;
+        this.userContexts = userContexts;
     }
 
     /**
-     * Returns the credentials used when the user associated with this session
-     * authenticated.
+     * Returns the authenticated user associated with this session.
      *
      * @return
-     *     The credentials used when the user associated with this session
-     *     authenticated.
+     *     The authenticated user associated with this session.
      */
-    public Credentials getCredentials() {
-        return credentials;
+    public AuthenticatedUser getAuthenticatedUser() {
+        return authenticatedUser;
     }
 
     /**
-     * Replaces the credentials associated with this session with the given
-     * credentials.
+     * Replaces the authenticated user associated with this session with the
+     * given authenticated user.
      *
-     * @param credentials
-     *     The credentials to associate with this session.
+     * @param authenticatedUser
+     *     The authenticated user to associated with this session.
      */
-    public void setCredentials(Credentials credentials) {
-        this.credentials = credentials;
+    public void setAuthenticatedUser(AuthenticatedUser authenticatedUser) {
+        this.authenticatedUser = authenticatedUser;
     }
-    
+
     /**
      * Returns the UserContext associated with this session.
      *
      * @return The UserContext associated with this session.
      */
     public UserContext getUserContext() {
```

Comment:
No, because that would break the build.

Would you prefer this be merged into an interim dev branch, so the enormous "DO NOT USE THIS SERIOUSLY" message is not ever on master?